tests/syscall/mount-overlayfs.bats

```diff
@@ -0,0 +1,278 @@
+    "mkdir -p $lower_path $upper_path $work_path $merge_path && \
+     mkdir -p $lower_dir $upper_dir && \
+     touch $lower_file $upper_file"
+  [ "$status" -eq 0 ]
+
+  # Mount overlayfs at $merge_path.
+  docker exec "$syscont" bash -c \
+    "mount -t overlay overlay -olowerdir=$lower_path,upperdir=$upper_path,workdir=$work_path $merge_path"
+  [ "$status" -eq 0 ]
+
+  verify_syscont_overlay_mnt $syscont $merge_path
+
+  # Verify dir/file content in merge folder is the expected one.
+  docker exec "$syscont" stat $lower_dir $lower_file $upper_dir $upper_file
+  [ "$status" -eq 0 ]
+
+  # Remove existing upper dir.
+  docker exec "$syscont" bash -c "rm -rf $merge_path/upper_dir"
+  [ "$status" -eq 0 ]
+
+  # Remove existing upper file.
+  docker exec "$syscont" bash -c "rm -rf $merge_path/upper_file"
+  [ "$status" -eq 0 ]
+
+  # Confirm removed upper dir/file have been eliminated.
+  docker exec "$syscont" bash -c "stat $merge_path/upper_dir || stat $merge_path/upper_file"
+  [ "$status" -eq 1 ]
+
+  # Remove existing lower dir.
+  docker exec "$syscont" bash -c "rm -rf $merge_path/lower_dir"
+  [ "$status" -eq 0 ]
+
+  # Remove existing lower file.
+  docker exec "$syscont" bash -c "rm -rf $merge_path/lower_file"
+  [ "$status" -eq 0 ]
+
+  # Confirm removed lower dir/file have been eliminated.
+  docker exec "$syscont" bash -c "stat $merge_path/upper_dir || stat $merge_path/upper_file"
+  [ "$status" -eq 1 ]
+
+  # Verify 'whiteout' has been created for removed dir.
+  docker exec "$syscont" bash -c "ls -l $upper_path/lower_dir"
+  [ "$status" -eq 0 ]
+  [[ $output =~ "c---------    1 root     root        0,   0".+"$upper_path/lower_dir" ]]
+
+  # Verify 'whiteout' has been created for removed file.
+  docker exec "$syscont" bash -c "ls -l $upper_path/lower_file"
+  [ "$status" -eq 0 ]
+  [[ $output =~ "c---------    1 root     root        0,   0".+"$upper_path/lower_file" ]]
+
+  # Umount overlayfs mountpoint.
+  docker exec "$syscont" bash -c "umount $merge_path"
+  [ "$status" -eq 0 ]
+
+  verify_syscont_overlay_umnt $syscont $merge_path
+
+  docker_stop "$syscont"
+}
+
+# Mix of various overlayfs mounts and umounts operations from chroot'ed contexts,
+# creation of whiteouts, etc -- fairly decent cocktail of ovs mount instructions.
+@test "mount overlayfs advanced ops" {
+
+  local syscont=$(docker_run --rm nestybox/alpine-docker-dbg:latest tail -f /dev/null)
+
+  docker exec -d "$syscont" sh -c "dockerd > /var/log/dockerd.log 2>&1"
+  [ "$status" -eq 0 ]
+
+  wait_for_inner_dockerd $syscont
+
+  docker exec -d "$syscont" sh -c "docker pull quay.io/coreos/flannel:latest 2>&1"
+  [ "$status" -eq 0 ]
+
+  docker_stop "$syscont"
+}
+
+# Verify overlayfs mounts with relative paths.
+@test "mount overlayfs relative paths" {
+
+  local lower_path="$ovs_path"/root/lower
+  local upper_path="$ovs_path"/root/upper
+  local work_path="$ovs_path"/root/work
+  local merge_path="$ovs_path"/root/merge
+
+  local rel_lower_dir=root/lower
```

Comment:
how about "./lower" (use the "." to force path resolution).

Reply:
Don't see the need for this.